### ISSUE TITLE
[YCQL] Malformed YCQL query triggers FATAL error

### ISSUE THREAD

**Issue by tvesely** [yugabyte/yugabyte-db#6879]
Running a YCQL query with square brackets around a column name triggers a FATAL server error. It looks like a parser error.

Reproduction steps:

```
ycqlsh> create keyspace repro;
ycqlsh> create table repro.test(id bigint PRIMARY KEY);
ycqlsh> select [id] from repro.test;
command terminated with exit code 137
```

Error from server logs:

```
I0115 00:03:39.449213   228 table_creator.cc:307] Created table repro.test of type YQL_TABLE_TYPE
I0115 00:03:49.812830   228 pt_expr.h:207] Missing QLName for expression(14) that is being selected
F0115 00:03:49.812961   228 eval_const.cc:98] Unknown datatype for QL constant value
Fatal failure details written to /mnt/disk0/yb-data/tserver/logs/yb-tserver.FATAL.details.2021-01-15T00_03_49.pid1.txt
F20210115 00:03:49 ../../src/yb/yql/cql/ql/exec/eval_const.cc:98] Unknown datatype for QL constant value
    @     0x7ff741f69ccc  yb::LogFatalHandlerSink::send()
    @     0x7ff74114e376  google::LogMessage::SendToLog()
    @     0x7ff74114b7da  google::LogMessage::Flush()
    @     0x7ff74114e8a9  google::LogMessageFatal::~LogMessageFatal()
    @     0x7ff74c2c3a4c  yb::ql::Executor::PTConstToPB()
    @     0x7ff74c2c1efc  yb::ql::Executor::PTExprToPB()
    @     0x7ff74c2c392e  yb::ql::Executor::PTConstToPB()
    @     0x7ff74c2c5a86  yb::ql::Executor::PTExprToPB()
    @     0x7ff74c2ec57b  yb::ql::Executor::ExecPTNode()
    @     0x7ff74c2f2621  yb::ql::Executor::ExecTreeNode()
    @     0x7ff74c2f2802  yb::ql::Executor::Execute()
    @     0x7ff74c2f29c8  yb::ql::Executor::ExecuteAsync()
    @     0x7ff74c74c5c9  yb::ql::QLProcessor::ExecuteAsync()
    @     0x7ff74c74c7de  yb::ql::QLProcessor::RunAsync()
    @     0x7ff74d512a4f  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7ff74d5174e8  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7ff74d51793a  yb::cqlserver::CQLProcessor::ProcessCall()
    @     0x7ff74d5321ec  yb::cqlserver::CQLServiceImpl::Handle()
    @     0x7ff7437c82b9  yb::rpc::ServicePoolImpl::Handle()
    @     0x7ff74376ed34  yb::rpc::InboundCall::InboundCallTask::Run()
    @     0x7ff7437da8b8  yb::rpc::(anonymous namespace)::Worker::Execute()
    @     0x7ff74200142f  yb::Thread::SuperviseThread()
    @     0x7ff73c821694  start_thread
    @     0x7ff73bf5e41d  __clone
    @              (nil)  (unknown)

*** Check failure stack trace: ***
    @     0x7ff741f680b1  yb::(anonymous namespace)::DumpStackTraceAndExit()
    @     0x7ff74114bc8d  google::LogMessage::Fail()
    @     0x7ff74114ddfd  google::LogMessage::SendToLog()
    @     0x7ff74114b7da  google::LogMessage::Flush()
    @     0x7ff74114e8a9  google::LogMessageFatal::~LogMessageFatal()
    @     0x7ff74c2c3a4c  yb::ql::Executor::PTConstToPB()
    @     0x7ff74c2c1efc  yb::ql::Executor::PTExprToPB()
    @     0x7ff74c2c392e  yb::ql::Executor::PTConstToPB()
    @     0x7ff74c2c5a86  yb::ql::Executor::PTExprToPB()
    @     0x7ff74c2ec57b  yb::ql::Executor::ExecPTNode()
    @     0x7ff74c2f2621  yb::ql::Executor::ExecTreeNode()
    @     0x7ff74c2f2802  yb::ql::Executor::Execute()
    @     0x7ff74c2f29c8  yb::ql::Executor::ExecuteAsync()
    @     0x7ff74c74c5c9  yb::ql::QLProcessor::ExecuteAsync()
    @     0x7ff74c74c7de  yb::ql::QLProcessor::RunAsync()
    @     0x7ff74d512a4f  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7ff74d5174e8  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7ff74d51793a  yb::cqlserver::CQLProcessor::ProcessCall()
    @     0x7ff74d5321ec  yb::cqlserver::CQLServiceImpl::Handle()
    @     0x7ff7437c82b9  yb::rpc::ServicePoolImpl::Handle()
    @     0x7ff74376ed34  yb::rpc::InboundCall::InboundCallTask::Run()
    @     0x7ff7437da8b8  yb::rpc::(anonymous namespace)::Worker::Execute()
    @     0x7ff74200142f  yb::Thread::SuperviseThread()
    @     0x7ff73c821694  start_thread
    @     0x7ff73bf5e41d  __clone
    @              (nil)  (unknown)
*** Aborted at 1610669029 (unix time) try "date -d @1610669029" if you are using GNU date ***
PC: @     0x7ff73beac5a6 __GI_abort
*** SIGSEGV (@0x0) received by PID 1 (TID 0x7ff72a54f700) from PID 0; stack trace: ***
    @     0x7ff73c829ba0 (unknown)
    @     0x7ff73beac5a6 __GI_abort
    @     0x7ff741f68104  yb::(anonymous namespace)::DumpStackTraceAndExit()
    @     0x7ff74114bc8c  google::LogMessage::Fail()
    @     0x7ff74114ddfc  google::LogMessage::SendToLog()
    @     0x7ff74114b7d9  google::LogMessage::Flush()
    @     0x7ff74114e8a8  google::LogMessageFatal::~LogMessageFatal()
    @     0x7ff74c2c3a4b  yb::ql::Executor::PTConstToPB()
    @     0x7ff74c2c1efb  yb::ql::Executor::PTExprToPB()
    @     0x7ff74c2c392d  yb::ql::Executor::PTConstToPB()
    @     0x7ff74c2c5a85  yb::ql::Executor::PTExprToPB()
    @     0x7ff74c2ec57a  yb::ql::Executor::ExecPTNode()
    @     0x7ff74c2f2620  yb::ql::Executor::ExecTreeNode()
    @     0x7ff74c2f2801  yb::ql::Executor::Execute()
    @     0x7ff74c2f29c7  yb::ql::Executor::ExecuteAsync()
    @     0x7ff74c74c5c8  yb::ql::QLProcessor::ExecuteAsync()
    @     0x7ff74c74c7dd  yb::ql::QLProcessor::RunAsync()
    @     0x7ff74d512a4e  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7ff74d5174e7  yb::cqlserver::CQLProcessor::ProcessRequest()
    @     0x7ff74d517939  yb::cqlserver::CQLProcessor::ProcessCall()
    @     0x7ff74d5321eb  yb::cqlserver::CQLServiceImpl::Handle()
    @     0x7ff7437c82b8  yb::rpc::ServicePoolImpl::Handle()
    @     0x7ff74376ed33  yb::rpc::InboundCall::InboundCallTask::Run()
    @     0x7ff7437da8b7  yb::rpc::(anonymous namespace)::Worker::Execute()
    @     0x7ff74200142e  yb::Thread::SuperviseThread()
    @     0x7ff73c821694 start_thread
    @     0x7ff73bf5e41d __clone
    @                0x0 (unknown)
```

**Comment by kmuthukk** [comment]
cc: @nocaway , @OlegLoginov

**Comment by m-iancu** [comment]
Fixed by 7182257, closing.
